llvm/lib/Transforms/Utils/SYCLUtils.cpp

```diff
@@ -0,0 +1,27 @@
+//===------------ SYCLUtils.cpp - SYCL utility functions ------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// SYCL utility functions.
+//===----------------------------------------------------------------------===//
+#include "llvm/Transforms/Utils/SYCLUtils.h"
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace llvm {
+
+void writeSYCLStringTable(const SYCLStringTable &Table, raw_ostream &OS) {
+  assert(Table.size() > 0 && "table should contain at least column titles");
+  size_t numberColumns = Table[0].size();
```

Comment:
This variable is used only in asserts, so it's unused in the builds with disabled assertions. I suggest remove it and use `Table[0].size()` in asserts instead.

Reply:
Removed the variable.